leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/CaliforniumEndpointsManager.java

```diff
@@ -191,6 +193,13 @@ public synchronized ServerIdentity createEndpoint(ServerInfo serverInfo) {
             } else {
                 throw new RuntimeException("Unable to create connector : unsupported security mode");
             }
+
+            // For bootstrap no need to have DTLS role exchange
+            // and so we can set DTLS Connection as client only by default.
+            if (serverInfo.bootstrap && incompleteConfig.isClientOnly() == null) {
+                newBuilder.setClientOnly();
+            }
```

Comment:
I ended up having auto detection from certificate what to configure:
https://github.com/dachaac/leshan/commit/fea11c87b0629a9e6ecf9a167b6d33c04bd28b2a

and the matching one for server:
https://github.com/dachaac/leshan/commit/613b6fbd7edd1fee8a4517d5f5e2ea0bde32f08b

Reply:
This changes is about the library itself and not really related to certificate.

Having LWM2M client and server which could act as both  DTLS client and  DTLS server  is mainly needed when DTLS connection is lost (reboot/crash/connection expired) and server initiated request. (for [more details](https://github.com/OpenMobileAlliance/OMA_LwM2M_for_Developers/issues/410))

In case of bootstrap session we don't need it, because session is short and always initiated by client. So I prefer to set the LWM2M client as DTLS clientOnly for bootstrap.

About "auto detection" this could be a good idea for demo, we add a warning explaining that failover will not work in case we provide a `clientAuth` or `serverAuth` only certificate.